gtsfm/densify/mvsnets/methods/PatchmatchNet/datasets/gtsfm.py

```diff
@@ -0,0 +1,131 @@
+        view_ids = [ref_view] + src_views[:self.nviews - 1]
+
+        imgs_0 = []
+        imgs_1 = []
+        imgs_2 = []
+        imgs_3 = []
+        depth_min = None
+        depth_max = None
+        proj_matrices_0 = []
+        proj_matrices_1 = []
+        proj_matrices_2 = []
+        proj_matrices_3 = []
+
+        for i, vid in enumerate(view_ids):
+            img = self.data['images'][vid]
+            np_img = np.array(img, dtype=np.float32) / 255.
+            np_img = cv2.resize(np_img, self.img_wh, interpolation=cv2.INTER_LINEAR)
+            
+            h, w, _ = np_img.shape
+                    
+            imgs_0.append(np_img)
+            imgs_1.append(cv2.resize(np_img, (w//2, h//2), interpolation=cv2.INTER_LINEAR))
+            imgs_2.append(cv2.resize(np_img, (w//4, h//4), interpolation=cv2.INTER_LINEAR))
+            imgs_3.append(cv2.resize(np_img, (w//8, h//8), interpolation=cv2.INTER_LINEAR))
+
+            intrinsics, extrinsics = self.data['cameras'][vid]
+            depth_min_, depth_max_ = self.data['depthRange'][0][i], self.data['depthRange'][1][i]
+
+            # multiply intrinsics and extrinsics to get projection matrix
+            proj_mat = extrinsics.copy()
+            intrinsics[:2,:] *= 0.125
+            proj_mat[:3, :4] = np.matmul(intrinsics, proj_mat[:3, :4])
+            proj_matrices_3.append(proj_mat)
+
+            proj_mat = extrinsics.copy()
+            intrinsics[:2,:] *= 2
+            proj_mat[:3, :4] = np.matmul(intrinsics, proj_mat[:3, :4])
+            proj_matrices_2.append(proj_mat)
+
+            proj_mat = extrinsics.copy()
+            intrinsics[:2,:] *= 2
+            proj_mat[:3, :4] = np.matmul(intrinsics, proj_mat[:3, :4])
+            proj_matrices_1.append(proj_mat)
+
+            proj_mat = extrinsics.copy()
+            intrinsics[:2,:] *= 2
+            proj_mat[:3, :4] = np.matmul(intrinsics, proj_mat[:3, :4])
+            proj_matrices_0.append(proj_mat)
+
+
+            if i == 0:  # reference view
+                depth_min = depth_min_
+                depth_max = depth_max_
+
+        imgs_0 = np.stack(imgs_0).transpose([0, 3, 1, 2])
+        imgs_1 = np.stack(imgs_1).transpose([0, 3, 1, 2])
+        imgs_2 = np.stack(imgs_2).transpose([0, 3, 1, 2])
+        imgs_3 = np.stack(imgs_3).transpose([0, 3, 1, 2])
+        imgs = {}
+        imgs['stage_0'] = imgs_0
+        imgs['stage_1'] = imgs_1
+        imgs['stage_2'] = imgs_2
+        imgs['stage_3'] = imgs_3
+        # proj_matrices: N*4*4
+        proj_matrices_0 = np.stack(proj_matrices_0)
+        proj_matrices_1 = np.stack(proj_matrices_1)
+        proj_matrices_2 = np.stack(proj_matrices_2)
+        proj_matrices_3 = np.stack(proj_matrices_3)
+        proj={}
+        proj['stage_3']=proj_matrices_3
+        proj['stage_2']=proj_matrices_2
+        proj['stage_1']=proj_matrices_1
+        proj['stage_0']=proj_matrices_0
+
+        return {"idx": idx,
```

Comment:
can you reformat all the files with `black gtsfm.py -l 120`?

https://github.com/psf/black